var/spack/repos/builtin/packages/spdk/package.py

```diff
@@ -0,0 +1,131 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import os
+
+from spack.package import *
+
+
+class Spdk(AutotoolsPackage):
+    """The Storage Performance Development Kit (SPDK) provides a set of tools
+    and libraries for writing high performance, scalable, user-mode storage
+    applications. It achieves high performance by moving all of the
+    necessary drivers into userspace and operating in a polled mode instead
+    of relying on interrupts, which avoids kernel context switches and
+    eliminates interrupt handling overhead.
+    """
+
+    homepage = "https://spdk.io"
+    url = "https://github.com/spdk/spdk/archive/v23.01.tar.gz"
+    git = "https://github.com/spdk/spdk"
+    maintainers("hyoklee")
+
+    version("master", branch="master", submodules=True)
+    version("23.01", tag="v23.01", submodules=True)
+    version("22.01.2", tag="v22.01.2", submodules=True)
+    version("21.07", tag="v21.07", submodules=True)
+    version("20.01.2", tag="v20.01.2", submodules=True)
+    version("20.01.1", tag="v20.01.1", submodules=True)
+    version("19.04.1", tag="v19.04.1", submodules=True)
+    version("19.04", tag="v19.04", submodules=True)
+    version("19.01.1", tag="v19.01.1", submodules=True)
+    version("19.01", tag="v19.01", submodules=True)
+    version("18.10.2", tag="v18.10.2", submodules=True)
+    version("18.10.1", tag="v18.10.1", submodules=True)
+    version("18.10", tag="v18.10", submodules=True)
+    version("18.07.1", tag="v18.07.1", submodules=True)
+    version("18.07", tag="v18.07", submodules=True)
+
+    variant("crypto", default=False, description="Build vbdev crypto module")
+    variant("fio", default=False, description="Build fio plugin")
+    variant("vhost", default=False, description="Build vhost target")
+    variant(
+        "virtio", default=False, description="Build vhost initiator and virtio-pci bdev modules"
+    )
```

Comment:
Given the description, should there be a conflict if `~vhost+virtio`?

Reply:
@tldahlgren , that's a good catch!
I thought it will fail but test passed with `~vhost+virtio`:

https://github.com/hyoklee/spack/actions/runs/4205727843/

@soumagne wrote the initial code for this package. 
Thus, I'd like to defer to @soumagne 's opinion.

Reply:
@mchaarawi, can you answer my question since @soumagne is not responding?

Reply:
i am not familiar with this. you better ask the spdk community: https://spdk.io/community/

Reply:
Thanks, @mchaarawi !

Reply:
@hyoklee sorry I had missed your earlier question. Same comment here as for the other packages, you want to remove the previous versions first and clean up the options. This package is particularly problematic because earlier versions were lacking install rules etc. The options passed here are based on what daos uses and some of these options only exist in specific spdk versions.